src/bot.rs

```diff
@@ -340,6 +350,7 @@ impl RcBot {
 
     /// helper function to start the event loop
     pub fn run<'a>(&'a self, core: &mut Core) -> Result<(), Error> {
+        *self.inner.name.borrow_mut() = Some(core.run(self.get_me().send()).unwrap().1.username.unwrap());
```

Comment:
Hey, this works for sure. Here is another way which doesn't block before the main run command.
``` rust
        // we want to use the get_me function here, therefore import the corresponding trait
        use functions::FunctionGetMe;

        // create a local copy of the bot to circumvent lifetime issues
        let bot = self.inner.clone();

        // create a new task which resolves the bot name and then set it in the struct
        let resolve_name = self.get_me().send()
            .map(move |user| {
                if let Some(name) = user.1.username {
                    bot.name.replace(Some(format!("@{}", name)));                                                                                                                            
                }
            }); 

        // spawn the task
        self.inner.handle.spawn(resolve_name.map_err(|_| ()));
```